docs/olmv1_mvp.md

```diff
@@ -0,0 +1,136 @@
+
+There is a dependency model in which extensions can rely on each other for required services that are out of scope of the primary purpose of the extension, allowing each extension to focus on a specific purpose. OLM also prevents conflicting extensions from running on the cluster, either with conflicting dependency constraints or conflicts in ownership of services provided via APIs.
+
+Since cluster extensions need to be supported with an enterprise-grade product lifecycle the specific scenarios in which an extension can be installed or updated are going to be limited by the author of the extension, primarily to align with what was tested by their QE processes. OLM allows the author to enforce these support  limitations in the form of additional constraints.
+
+During their lifecycle on the cluster, OLM also manages the permissions and capabilities extensions have on the cluster as well as the permission and access tenants on the cluster have to the extensions. This is done using the Kubernetes RBAC system in combination with tenant isolation using Kubernetes namespaces. The interaction surface of the extensions is solely composed of Kubernetes APIs the extensions define.
+
+OLM also defines a packaging model in which catalogs of extensions, usually containing the entire version history of each extension, are made available to clusters for cluster users to browse and select from. Via new versions of extensions delivered with this packaging system, OLM is able to apply updates to existing running extensions on the cluster in a way where the integrity of the cluster is maintained and constraints and dependencies are kept satisfied.
+
+The scope of all these features is a single cluster so far with namespace-based handling of catalog access and extension API accessibility and discoverability. Expansion of this scope is indirectly expected through the work of the Kubernetes Control Plane (kcp) project, which in its first incarnation will likely use its own synchronization mechanism to get OLM-managed extensions deployed eventually on one or more physical clusters from a shared, virtual control plane called a “workspace”. This is an area under active development and subject to change, OLM might need to become aware of kcp in a future state.
+
+**Verdict:** The purpose of OLM remains the same. The scope of OLM will increase to span multiple clusters following the kcp model, though likely many aspects of this will become transparent to OLM itself through the workspace abstraction that kcp provides. What needs to change in OLM 1.0 is how the above tasks are carried out from the user perspective and how much control users have in the process and which persona is involved.
+
+## Functional Requirements
+_Priority Rating: 1 highest, 2 medium, 3 lower (e.g. P2 = Medium Priority)_
+
+**F1 - Extension catalogs (P1):** The existing OLM concepts around catalogs, packages and channels is to be used as a basis for below functional requirements.
```

Comment:
From a contributor perspective, I think it would be neat to set these up as check boxes so that when we have met these requirements we can check them off. Would be helpful to easily tell which things have been achieved and which have not.

Reply:
I like the idea of having a visual checklist to track progress, but at the same time, I worry the rollout of certain items may occur in multiple phases, which could make it difficult to promptly check items off.  As such, should we leave the bullet list as is for the time being and submit a pull request for an update once there are features that can be checked off?

Reply:
IMO an item should not be checked off until it is fully implemented. I'm fine with checking off after the fact if we are okay with that

Reply:
I would at least make the individual items markdown headers so that we get link anchors out of them. But that can easily happen in a follow-up.